src/components/my-career/SearchJobRole.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { getConfig } from '@edx/frontend-platform/config';
+import { logError } from '@edx/frontend-platform/logging';
+import { InstantSearch } from 'react-instantsearch-dom';
+import { SearchContext, deleteRefinementAction } from '@edx/frontend-enterprise-catalog-search';
+import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
+import FacetListRefinement from '@edx/frontend-enterprise-catalog-search/FacetListRefinement';
+import {
+  Button, Form, StatefulButton,
+} from '@edx/paragon';
+import { CURRENT_JOB_FACET } from '../skills-quiz/constants';
+import { patchProfile, fetchJobDetailsFromAlgolia } from './data/service';
+import { CURRENT_JOB_PROFILE_FIELD_NAME, SAVE_BUTTON_LABELS } from './data/constants';
+import { useAlgoliaSearch } from '../../utils/hooks';
+
+const SearchJobRole = (props) => {
+  const config = getConfig();
+  const [searchClient, searchIndex] = useAlgoliaSearch(config, config.ALGOLIA_INDEX_NAME_JOBS);
+  const { username } = getAuthenticatedUser();
+  const { refinements, dispatch } = useContext(SearchContext);
+  const { current_job: currentJob } = refinements;
+  const [loadingRequest, setLoadingRequest] = useState(false);
+  const [edited, setEdited] = useState(false);
+  const [requestComplete, setRequestComplete] = useState(false);
+  const [requestError, setRequestError] = useState(false);
+
+  const {
+    title, attribute, typeaheadOptions, facetValueType, customAttribute,
+  } = CURRENT_JOB_FACET;
+
+  useEffect(() => {
+    if (currentJob) {
+      setEdited(true);
+    }
+  }, [currentJob]);
+
+  const getButtonState = () => {
+    if (requestError) {
+      return 'error';
+    }
+
+    if (requestComplete) {
+      return 'complete';
+    }
+
+    if (loadingRequest) {
+      return 'pending';
+    }
+
+    if (edited) {
+      return 'edited';
+    }
+
+    return 'default';
+  };
+
+  const handleSubmit = async () => {
+    let resp = {};
+    setLoadingRequest(true);
+    const { id: currentJobID } = await fetchJobDetailsFromAlgolia(searchIndex, currentJob);
+    const params = {
+      extended_profile: [
+        { field_name: CURRENT_JOB_PROFILE_FIELD_NAME, field_value: currentJobID },
+      ],
+    };
+
+    try {
+      resp = await patchProfile(username, params);
+    } catch (error) {
+      setLoadingRequest(false);
+      setRequestError(true);
+      logError(new Error(error));
+    }
+    setRequestComplete(true);
+    setLoadingRequest(false);
+    setEdited(false);
+    if (edited) {
+      dispatch(deleteRefinementAction(customAttribute));
+    }
+    props.onSave(resp);
```

Comment:
Question: Can you move this line right below [this line](https://github.com/openedx/frontend-app-learner-portal-enterprise/pull/769/files#diff-09102a0bcd64420bed56c2021f674f11972e0943869b517a5fd240d6424c20fcR69) that way you won't have to defined `resp` outside the `try..catch` block.

Reply:
This won't be possible because the method "onSave" remove this component from DOM and I want to smoothly transition from this component to next one.